sdk/core/System.ClientModel/src/Convenience/ClientConnection.cs

```diff
@@ -19,11 +18,17 @@ public readonly struct ClientConnection
     public ClientConnection(string id, string locator, object credential, CredentialKind credentialKind)
```

Comment:
we should remove this whole implementation and simply call the longer ctor passing in null as the metadata.